doc/source/Parameters.rst

```diff
@@ -166,9 +190,25 @@ For all on/off integer flags, 0 is off and 1 is on.
 
 .. c:var:: int photoelectric_heating
 
-   Flag to enable a spatially uniform heating term approximating
-   photo-electric heating from dust from `Tasker & Bryan (2008)
-   <http://adsabs.harvard.edu/abs/2008ApJ...673..810T>`_.  Default: 0.
+   Flag to enable photo-electric heating from irradiated dust grains.
+   Default: 0.
+
+    - 0: no photo-electric heating.
+    - 1: a spatially uniform heating term from `Tasker & Bryan (2008)
+      <http://adsabs.harvard.edu/abs/2008ApJ...673..810T>`__. The exact
+      heating rate used must be specified with the
+      :c:data:`photoelectric_heating_rate` parameter. For temperatures
+      above 20,000 K, the photo-electric heating rate is set to 0.
+    - 2: similar to option 1, except the heating rate is calculated
+      using equation 1 of `Wolfire et al. (1995)
+      <https://ui.adsabs.harvard.edu/abs/1995ApJ...443..152W/abstract>`__
+      and the user must supply the intensity of the interstellar radiation
+      field with the :c:data:`interstellar_radiation_field` parameter. The
+      value of epsilon is taken as a constant equal to 0.05 for gas below
+      20,000 K and 0 otherwise.
+    - 3: similar to option 1, except the value of epsilon is calculated
+      directly from equation 2 of `Wolfire et al. (1995)
+      <https://ui.adsabs.harvard.edu/abs/1995ApJ...443..152W/abstract>`__.
```

Comment:
I'm not sure if we need to be in the business of making sure we've appropriately added all caveats to every piece of code, but after looking at this again I think it may be worth adding a word of caution when using `photoelectric_heating=3` in simulations that go down to very low H/He ionization fractions / low temperatures. 

The electron number density is likely to be dominated from contributions from metals (C and O), PAHs, and grains in this regime, which are not tracked by any of the chemistry solvers in Grackle at the moment. This would tend to underestimate (possibly significantly) the efficiency, weakening photoelectric heating in regions where it could likely be a dominant heating source. 

I'm not sure exactly how big of an issue this can be in a real simulation, but I think it could be worth either: 1) mentioning this potential issue, or 2) including some sort of electron number density fraction (n_e/n) or efficiency floor in case n_e becomes unrealistically low.

Reply:
It's a good point and something we could probably improve on. With `primordial_chemistry>0`, we don't make use of the tabulated mean molecular weight data from Cloudy, but we probably could do to make a slightly improved estimate of the electron contribution from some of the metals. This is a bigger task than I'd like to take on for this PR, but maybe in the future. For now, I'll go ahead and add a caveat in the docs.